security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -630,12 +637,29 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 		if sc.shouldRotate(&secret) {
 			atomic.AddUint64(&sc.secretChangedCount, 1)
 			// Send the notification to close the stream if token is expired, so that client could re-connect with a new token.
-			if sc.isTokenExpired(&secret) {
-				cacheLog.Debugf("%s token expired", logPrefix)
+			isTokenExpired := sc.isTokenExpired(&secret)
```

Comment:
I don't think you need to change the token expiration check logic here. There should be a flag passed in, say, "supportRotationWithCert". If the flag is set to true, you should use cert, instead of token to rotate.

Reply:
what if the cert is expired or not valid. In my understanding if it is not valid we switch to use the token. and when the token is expired, we will not use the cert instead and just close the stream. Is that the case?